internal/statemachine/classic_states.go

```diff
@@ -252,6 +259,22 @@ func (stateMachine *StateMachine) calculateStates() error {
 	// This includes the no-op "finish" state to signify successful setup
 	rootfsCreationStates = append(rootfsCreationStates, imageCreationStates...)
 
+	// only run makeDisk if there is an artifact to make
+	if classicStateMachine.ImageDef.Artifacts.Img != nil {
+		rootfsCreationStates = append(rootfsCreationStates,
+			stateFunc{"make_disk", (*StateMachine).makeDisk})
```

Comment:
Proposition: could we rename the `make_disk` state to `make_img`/`make_raw` or something? Since there might be more disk formats, and I'd assume we'd want separate states to handle those.

Reply:
This sounds like something to do when new disk formats are added.